.gitlab-ci.yml

```diff
@@ -337,12 +336,15 @@ unleash-check:
   <<:                              *test-refs-no-trigger
```

Comment:
do you really want to publish before the tests are green?

Reply:
That's the `unleash-check` - I think it should be run early/at the same time as tests; only publishing should be done after every test is green

Reply:
In my view (I actually do not know what does it check, `Check whether crates can be packaged` in the docs is probably not enough) it's an overkill to run these checks on every commit. We should run them before the release target, so maybe on master, or even nightly?

Reply:
If I understand this correctly, what you propose may lead to the PR checks being green at first but only discovered to be red once merged. If that's the case, then won't be much incentive to fix this once a PR is merged and it'll just hang around being red until someone disables it again - I'd rather we catch that early.

If this is deemed to expensive to run on every commit, we might want to run this as an extra step for `bot merge` to not merge a given PR until we know for sure that this check is green.

Reply:
It is expensive, [the job](https://gitlab.parity.io/parity/substrate/-/jobs/1027034) takes 30 minutes. Any possible way to combine it with something to reuse the cache?

Ideally, it will run on the pre-merge pipeline and a merge-train, but they are not ready yet.

Reply:
Maybe it's better to either check only if certain files are changed or versions got bumped?